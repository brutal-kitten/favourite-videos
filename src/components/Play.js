import React, { Component } from 'react';

class Play extends Component {

  handleClick = (event) => {
    event.preventDefault();
    this.props.playVideo(this.props.id);
  }

  render() {
    return(
      <div className="play" >
        <button type="button" onClick={(event) => this.handleClick(event)}><span className="glyphicon glyphicon-play-circle"></span></button>
      </div>
    )
  }
}

export default Play;
