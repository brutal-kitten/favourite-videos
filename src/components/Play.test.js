import React from 'react';
import ReactDOM from 'react-dom';
import Play from './Play.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display a button', () => {
  const tree = TestRenderer.create(
    <Play />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleclick function after click', () => {
  const mockPlayVideo = jest.fn();
  const props = {
    playVideo: mockPlayVideo,
    id: "12345"
};

  const wrapper = shallow(<Play {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {}
  });

  it('should call playVideo function with 1 argument - id', () => {
    expect(mockPlayVideo.mock.calls.length).toBe(1);
    expect(mockPlayVideo.mock.calls[0][0]).toBe("12345");
  });
})
