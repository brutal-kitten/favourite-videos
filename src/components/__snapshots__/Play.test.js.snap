// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="play"
>
  <button
    className="buttonPlay"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="glyphicon glyphicon-play-circle"
    />
  </button>
</div>
`;
